### DISCUSSION TITLE
How to listen to every new block

### DISCUSSION THREAD

**Discussion by shivam2320** [aptos-labs/aptos-developer-discussions#198] [General]
I want to write a program which does some stuff on every new block
How can I listen to every new block

**Comment by heliuchuan** [comment]
To get the latest block you can use the type script SDK

`aptos.getLedgerInfo()`

will give you something like
```
  {
     "chain_id": 4,
     "epoch": "8",
     "ledger_version": "714",
     "oldest_ledger_version": "0",
     "ledger_timestamp": "1694695496521775",
     "node_role": "validator",
     "oldest_block_height": "0",
     "block_height": "359",
     "git_hash": "c82193f36f4e185fed9f68c4ad21f6c6dd390c6e"
    }
```

you can use that to get the latest block and query it via `aptos.getBlockByHeight({ blockHeight: number })`


Right now the SDK doesn't have any built in functionality to listen to new blocks, but you can write your own polling to do so.

**Comment by kaouki99** [comment]
You can connect via gRPC to stream the transactions https://aptos.dev/indexer/txn-stream/

**Reply by heliuchuan** [reply]
ah yes
